clients/js/src/chains/sei/submit.ts

```diff
@@ -131,7 +137,26 @@ export const submit = async (
   });
   const [account] = await wallet.getAccounts();
   const client = await getSigningCosmWasmClient(rpc, wallet);
-  const fee = calculateFee(300000, "0.1usei");
+
+  const executeContractMsg: MsgExecuteContractEncodeObject = {
+    typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
+    value: MsgExecuteContract.fromPartial({
+      sender: account.address,
+      contract: target_contract,
+      msg: toUtf8(JSON.stringify(execute_msg)),
+      funds: [],
+    }),
+  };
+  // For some reason, the simulation only provides the gas used but no events
+  // so we can't determine whether it worked unless we do away with cosmjs and query the node ourselves.
+  // See https://github.com/cosmos/cosmjs/issues/1148#issuecomment-1129259646
+  const gasUsed = await client.simulate(account.address, [executeContractMsg], undefined);
+  // It looks like the simulation is a bit lacking when it comes to estimating gas.
+  // See https://github.com/cosmos/cosmos-sdk/issues/4938
+  // That's why we multiply it by a factor of 1.3
+  const estimatedGas = Math.floor(gasUsed * 130 / 100);
+
+  const fee = calculateFee(estimatedGas, "0.1usei");
```

Comment:
should the 0.1 usei be pulled from somewhere?

Reply:
Absolutely no idea. It just worked for me, but it could easily be something else for all I know :P

Reply:
🤷